File: engine/engine-model/src/main/scala/tools/packet/buff/CancelBuff.scala
package tools.packet.buff

import client.MapleBuffStat
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class CancelBuff(private var _statIncreases: java.util.List[MapleBuffStat]) extends PacketInput {
  def statIncreases: java.util.List[MapleBuffStat] = _statIncreases

  override def opcode(): SendOpcode = SendOpcode.CANCEL_BUFF
}

File: engine/engine-model/src/main/java/tools/packet/buff/CancelBuff.java
package tools.packet.buff;

import java.util.List;

import client.MapleBuffStat;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record CancelBuff(List<MapleBuffStat> statIncreases) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.CANCEL_BUFF;
   }
}
